ckanext/versions/tests/test_actions.py

```diff
@@ -211,6 +213,166 @@ def test_resource_version_create_creator_user_id_parameter(self):
+        context = get_context(user)
+        version = resource_version_create(
+            context, {
+                'resource_id': resource['id'],
+                'name': '1.0',
+                'notes': 'Initial notes.'
+            }
+        )
+
+        new_creator = factories.User()
+        resource_version_update(
+            context, {
+                'version_id': version['id'],
+                'name': '2.0',
+                'notes': 'Updated notes.',
+                'creator_user_id': new_creator['id']
+            }
+        )
+        version = version_show(
+            context, {'version_id': version['id']}
+        )
+
+        assert version['name'] == '2.0'
+        assert version['notes'] == 'Updated notes.'
+        assert version['creator_user_id'] == new_creator['id']
+
+    def test_resource_version_update_overrides_if_not_provided(self):
+        resource = factories.Resource()
+        user = factories.Sysadmin()
+        context = get_context(user)
+        version = resource_version_create(
+            context, {
+                'resource_id': resource['id'],
+                'name': '1.0',
+                'notes': 'Initial notes.'
+            }
+        )
+
+        new_creator = factories.User()
+        context = get_context(new_creator)
+        resource_version_update(
+            context, {
+                'version_id': version['id'],
+                'name': '2.0',
+            }
+        )
+        version = version_show(
+            context, {'version_id': version['id']}
+        )
+
+        assert version['name'] == '2.0'
+        assert version['notes'] == None
+        assert version['creator_user_id'] == new_creator['id']
+
+    def test_name_must_be_unique(self):
+        resource = factories.Resource()
+        user = factories.Sysadmin()
+        context = get_context(user)
+        resource_version_create(
+            context, {
+                'resource_id': resource['id'],
+                'name': '1.0',
+                'notes': 'Initial notes.'
+            }
+        )
+
+        with pytest.raises(toolkit.ValidationError):
+            resource_version_create(
+            context, {
+                'resource_id': resource['id'],
+                'name': '1.0',
+                'notes': 'Another version 1.0'
+            }
+        )
+
+    def test_version_id_is_mandatory_for_update(self):
+        with pytest.raises(toolkit.ValidationError) as e:
+            resource_version_update({}, {'name': '2.0'})
+        assert 'Missing value' in str(e)
+        assert 'version_id' in str(e)
+
+    def test_version_name_is_mandatory_for_update(self):
+        with pytest.raises(toolkit.ValidationError) as e:
+            resource_version_update({}, {'version_id': 'fake-id'})
+        assert 'Missing value' in str(e)
+        assert 'name' in str(e)
+
+
+@pytest.mark.usefixtures('clean_db', 'versions_setup')
+class TestResourceVersionPatch(object):
```

Comment:
It's good to have at least one test using `helpers.call_action` or `toolkit.get_action` to no miss issues like the comment above